From ff56feab5a824cba486b5b03780c5339304aa87a Mon Sep 17 00:00:00 2001
From: Bernardo Rufino <brufino@google.com>
Date: Tue, 3 Oct 2017 13:55:10 +0100
Subject: [PATCH 002/306] Canonicalize notification channel sounds for backup

Canonicalize for backup and canonicalize and uncanonicalize for restore
(see comment).

Test: Set custom notification sound, make backup, remove notification
sound from device (from Ringtones and make sure to update media content
provider), restore => Observe default instead of random number. Do the
same without removing the sound and observe restores successfully.
Test: runtest systemui-notification
Bug: 66444697

(cherry picked from commit c27bb6ad34fd59161f6460b692ba72c7ede789b6)

Change-Id: I32c186d0d7479b01f6cc67cce9bc5cb66264a064
(cherry picked from commit 2d7a4a3f674d62adfb2825b738f2e87ec246194d)
---
 core/java/android/app/NotificationChannel.java | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/java/android/app/NotificationChannel.java b/core/java/android/app/NotificationChannel.java
index 47063f0842e..c06ad3f32cf 100644
--- a/core/java/android/app/NotificationChannel.java
+++ b/core/java/android/app/NotificationChannel.java
@@ -32,6 +32,8 @@ import android.util.proto.ProtoOutputStream;
 
 import com.android.internal.util.Preconditions;
 
+import com.android.internal.util.Preconditions;
+
 import org.json.JSONException;
 import org.json.JSONObject;
 import org.xmlpull.v1.XmlPullParser;
-- 
2.17.1

